src/Modules/UngradedModules/ModuleGens.jl

```diff
@@ -100,6 +100,37 @@ function singular_generators(M::ModuleGens)
   return M.S
 end
 
+@doc raw"""
+    singular_ordering(M::ModuleGens)
+
+Return the ordering of `M` from Singular side.
+"""
+function singular_ordering(M::ModuleGens)
+    singular_assure(M)
+    return Singular.ordering(base_ring(M.SF))
```

Comment:
```suggestion
    return Singular.ordering(base_ring(singular_freemodule(M)))
```
This could even get rid of one more assure call